Service/ZGWService.php

```diff
@@ -405,12 +435,73 @@ public function downloadInhoudHandler(array $data, array $configuration): array
     }//end downloadInhoudHandler()
 
 
+
+    /**
+     * Handles gebruiksrecht POST logic.
+     *
+     * @param array $data          The data passed by the action.
+     * @param array $configuration The configuration of the action.
+     * 
+     * @return array Http response.
+     */
+    public function gebruiksrechtHandler(array $data, array $configuration): array
```

Comment:
This function is looking kinda big as well.

Here i would try to combine all checks for possible errors into one function as well.